src/warc_types.rs

```diff
@@ -180,11 +181,190 @@ impl<R: BufRead> Iterator for WarcReader<R> {
             version: version_ref.to_owned(),
             headers: headers_ref
                 .into_iter()
-                .map(|(token, value)| (token.to_lowercase().to_owned(), value.to_owned()))
+                .map(|(token, value)| (token.into(), value.to_owned()))
                 .collect(),
             body: body_ref.to_owned(),
         };
 
+        // certain headers are required by the format
```

Comment:
This is cool and I'm glad to see it. I'm not sure we should make this part of the iterator though.

One thought would be to add a `verify` method to a warc record and let consumers choose to call this as needed during consumption or creation.

My thinking is, if I have warc file that doesn't follow the standard, as-is this would prevent me from getting at data that would otherwise be available.

Reply:
Hmm. I generally expect streams to only give me valid data. Then again, there is not currently a guarantee of validity on the `Record` type, as all its fields are public and it is fairly loosely defined.

I will implement `verify` as you suggest for this PR, and express my expectations as part of my refactoring (which is getting more and more major) to fix #13.